game-core/src/main/java/games/strategy/engine/posted/game/pbf/NodeBbForumPoster.java

```diff
@@ -106,12 +113,12 @@ public static NodeBbForumPoster newInstanceByName(final String name, final int t
    * @return true if the post was successful
    */
   public CompletableFuture<String> postTurnSummary(
-      final String summary, final String title, final Path path) {
+      final String summary, final String title, @Nullable final SaveGameParameter saveGame) {
```

Comment:
I'm not sure if parameter object is the right pattern here. I suspect factory methods might be more appropriate.

We have two invocations of this method:

```
                .postTurnSummary(
                    (gameNameAndInfo + "\n\n" + turnSummary),
                    "TripleA " + title + ": " + currentPlayer.getName() + " round " + roundNumber,
                    SaveGameParameter.builder()
                        .path(saveGameFile.toPath())
                        .displayName(saveGameName)
                        .build());
```
```
                      "Test summary from TripleA, engine version: "
                          + ClientContext.engineVersion()
                          + ", time: "
                          + DateTimeUtil.getLocalizedTime(),
                      "Testing Forum poster",
                      f != null
                          ? SaveGameParameter.builder()
                              .path(f.toPath())
                              .displayName("Test.jpg")
                              .build()
                          : null);
```

Having the nullable parameter value is a big code smell. 



I think the two APIs probably should be something more like:

```
              final CompletableFuture<String> future =
                  poster.postTestSummary();
```

And: 
```
              final CompletableFuture<String> future =
                  poster.post(gameData, historyLog, saveGameFile);
```

-----------

The odd thing about a parameter object here is that it's not capturing all of the parameters. It's really just a value object.

It would also be cool to see some of this logic extracted yet further and then tested so we can avoid such a regression going forward.

Reply:
I see your point. The reason I decided to implement it like this is to ensure that we don't end up in a state where for some reason one parameter is null and the other isn't

Reply:
Makes sense, any thoughts to rename the object? Any thoughts on how we might be able to test this? I suspect the test decision would drive most of the rest of the choices.

Side-note, given we have save game in so many places, it's really quite the true violation of DRY.